src/main/scala/io/github/pauljamescleary/petstore/domain/pets/PetService.scala

```diff
@@ -17,17 +17,11 @@ class PetService[F[_]](repository: PetRepositoryAlgebra[F], validation: PetValid
   import cats.syntax.all._
 
   def create(pet: Pet)(implicit M: Monad[F]): EitherT[F, PetAlreadyExistsError, Pet] =
-    for {
-      _ <- validation.doesNotExist(pet)
-      saved <- EitherT.liftF(repository.put(pet))
-    } yield saved
+    validation.doesNotExist(pet) *> EitherT.liftF(repository.create(pet))
```

Comment:
Would prefer to use for comprehension syntax and avoid symbolic operators when possible

Reply:
I'm trying to apply some judicious use of the other typeclasses besides just `Functor` and `Monad`.

In this case, we really do mean that the output of the left hand side is not passed to the right hand side. That is, the inner type need not be unwrapped from the context in order to process the following expression inside of the same context. `flatMap`, although more familiar to non-cats users, seems imprecise to me because of this. When one uses `flatMap`, one indicates that the inner object must be unwrapped and passed into the following computation.

I would argue in favor of using `*>` for this case, with a comment and a link to what this means, or we could switch to `productR`?  Alternatively we could use `productR` since it's easier to look up, with a comment perhaps that it's aliased as `*>`?  

What do you think?

Reply:
The other *Service classes have several methods that could be rewritten using `*>`. It reads like a hurdle. If the validation passes over the hurdle, provide the following expression.

Reply:
`productR` might be harder to understand ;)

Thanks for the explanation, that helps me alot.  I really like the hurdle analogy.

Could you add an inline comment when it is used to add context?  

And I would say if you use it in one place, use it in all the places where it makes sense for consistency.

Thanks for the clarification

Reply:
I've looked at the *> usage again, and I reverted it.  I think you're right about the readability. I have another comment in the main thread.